packages/ckeditor5-cloud-services/src/cloudservices.js

```diff
@@ -67,9 +76,61 @@ export default class CloudServices extends ContextPlugin {
 			return;
 		}
 
-		this.token = new CloudServices.Token( this.tokenUrl );
+		return this.registerTokenUrl( this.tokenUrl, true );
+	}
+
+	/**
+	 * Registers an additional authentication token URL for CKEditor Cloud Services or a callback to the token value promise. See the
+	 * {@link module:cloud-services/cloudservices~CloudServicesConfig#tokenUrl} for more details.
+	 *
+	 * @param {String|Function} tokenUrl The authentication token URL for CKEditor Cloud Services or a callback to the token value promise.
+	 * @param {Boolean} [isDefault=false] Whether this should be a default authentication token provider.
+	 * @returns {Promise.<module:cloud-services-core/token~Token>}
+	 */
+	registerTokenUrl( tokenUrl, isDefault = false ) {
+		if ( this._tokens.has( tokenUrl ) ) {
+			return Promise.resolve( this.getTokenFor( tokenUrl ) );
```

Comment:
Should not it throw an error here?

Reply:
Nope, it's reusing `Token` instance registered by the other feature - same `tokenUrl` = same token.

Reply:
I'll add code comment there

Reply:
Ok. This makes sense if two different features share the same configuration.